src/packages/@ncigdc/modern_components/ProjectPrimarySitesTable/ProjectPrimarySitesTable.model.js

```diff
@@ -0,0 +1,270 @@
+  var dataColumns = Object.keys(DATA_CATEGORIES).reduce((acc, k) => {
+    const type = props.repository.cases.aggregations.files__data_category.buckets.find(
+      item => item.key === DATA_CATEGORIES[k].full,
+    );
+    const linkQuery = {
+      filters: makeFilter([
+        {
+          field: 'cases.project.project_id',
+          value: props.projectId,
+        },
+        { field: 'files.data_category', value: DATA_CATEGORIES[k].full },
+        {
+          field: 'cases.primary_site',
+          value: props.primarySite,
+        },
+      ]),
+    };
+    return acc.concat(
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'flex-end',
+          alignItems: 'center',
+          minWidth: '80px',
+        }}
+        key={k}
+      >
+        {type ? (
+          <RepositoryFilesLink query={linkQuery}>
+            {type.doc_count}{' '}
+          </RepositoryFilesLink>
+        ) : (
+          <span>-- </span>
+        )}
+
+        <abbr
+          style={{
+            fontSize: '1rem',
+            fontVariantPosition: 'super',
+            paddingLeft: '5px',
+          }}
+        >
+          <Tooltip
+            Component={DATA_CATEGORIES[k].full}
+            style={tableToolTipHint()}
+          >
+            {DATA_CATEGORIES[k].abbr}
+          </Tooltip>
+        </abbr>
+      </div>,
+    );
+  }, []);
+  return (
+    <span
+      style={{
+        display: 'flex',
+        flexDirection: 'row',
+        justifyContent: 'space-between',
+      }}
+    >
+      {dataColumns}
+    </span>
+  );
+});
+
+let DiseaseList = withData(props => {
+  const { buckets } = props.repository.cases.aggregations.disease_type;
+  const diseasesByName = buckets.map(type => type.key).sort();
+  return (
+    <span>
+      {diseasesByName.length > 1 && (
+        <CollapsibleList
+          liStyle={{ whiteSpace: 'normal', listStyleType: 'disc' }}
+          toggleStyle={{ fontStyle: 'normal' }}
+          data={diseasesByName.slice(0).sort()}
+          limit={0}
+          expandText={`${diseasesByName.length} Disease Types`}
+          collapseText="collapse"
+        />
+      )}
+      {diseasesByName.length <= 1 && diseasesByName[0]}
+    </span>
+  );
+});
+
+let FilesByPrimarySite = withData(props => {
+  return (
+    <span
+      style={{
+        display: 'flex',
+        justifyContent: 'flex-end',
+        minWidth: '60px',
+      }}
+    >
+      <RepositoryFilesLink
+        query={{
+          filters: makeFilter([
+            {
+              field: 'cases.project.project_id',
+              value: props.projectId,
+            },
+            {
+              field: 'cases.primary_site',
+              value: props.primarySite,
+            },
+          ]),
+        }}
+      >
+        {props.repository.cases.aggregations.disease_type.buckets.length
+          ? props.repository.cases.aggregations.disease_type.buckets[0]
+              .doc_count
+          : '0 --'}
```

Comment:
should this just be `--` or 0 depending on if its missing or 0 length

Reply:
I checked, this was actually supposed to be just '0' if no files present.